src/miral/x11_support.cpp

```diff
@@ -34,13 +36,42 @@ miral::X11Support::X11Support() : self{std::make_shared<Self>()}
 
 void miral::X11Support::operator()(mir::Server& server) const
 {
+    static auto const x11_displayfd_opt = "x11-displayfd";
+
     server.add_configuration_option(
         mo::x11_display_opt,
         "Enable X11 support", mir::OptionType::null);
 
     server.add_configuration_option(
         "xwayland-path",
         "Path to Xwayland executable", "/usr/bin/Xwayland");
+
+    server.add_configuration_option(
+        x11_displayfd_opt,
+        "file descriptor to write X11 DISPLAY number to when ready to connect", mir::OptionType::integer);
+
+    server.add_init_callback([this, &server]
+        {
+            auto const options = server.get_options();
+
+            if (options->is_set(x11_displayfd_opt))
+            {
+                mir::Fd const fd{options->get<int>(x11_displayfd_opt)};
+
+                server.the_main_loop()->enqueue(this, [fd, &server]
+                    {
+                        if (auto const x11_display = server.x11_display())
+                        {
+                            auto const display = x11_display.value().substr(1) + "\n";
```

Comment:
Can you please document why you are chopping off the first character or, better yet, assert the first character is indeed a `:`